server/src/storage_engine/default_engine.rs

```diff
@@ -237,16 +156,59 @@ impl StorageEngine for DefaultEngine {
         Ok(())
     }
 
-    fn delete_file(&self, path: String) -> Result<(), EngineError> {
+    fn delete_directory_recursive(&self, path: String) -> Result<(), EngineError> {
         self.file_attr_db.db.delete(path.as_bytes())?;
+        self.delete_dir_recursive(path.clone())?;
+        self.delete_from_parent(path)?;
+        Ok(())
+    }
 
-        let index = match path.rfind('/') {
-            Some(value) => value,
-            None => return Err(EngineError::Path),
-        };
-        let (parent_dir, file_name) = (&path[..=index], &path[(index + 1)..]);
+    fn is_exist(&self, path: String) -> Result<bool, EngineError> {
```

Comment:
Returning `Result<(), EngineError>` might be enough.

Reply:
The `Result` of this function has different handling when used by `create` and `delete`, I don't think it's good to return `EngineError` directly

Reply:
fine